odoo/addons/test_testing_utilities/tests/test_env.py

```diff
@@ -0,0 +1,46 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo.tests.common import SavepointCase
+
+
+class TestEnv(SavepointCase):
+
+    @classmethod
+    def setUpClass(cls):
+        super(TestEnv, cls).setUpClass()
+        user = cls.env['res.users'].create({
+            'name': 'superuser',
+            'login': 'superuser',
+            'password': 'superuser',
+            'groups_id': [(6, 0, cls.env.user.groups_id.ids)],
+        })
+        cls.env = cls.env(user=user)
+
+        # Using the sudo mode, we create a new environment with the flag `su` set
+        # to True. That way, when using again the sudo mode in the tests, we will
+        # switch into that same environment
+        cls.env['res.company'].sudo().create({'name': 'Useless Company'})
```

Comment:
Instead of a happy side effect (which is not even deterministic), let's make the sudoed environment alive explicitly with
```suggestion
        # make sure there is at least another environment in the current transaction
        cls.sudo_env = cls.env(su=True)
```
We could actually make _any_ other environment, like `cls.env(context={'unexpected_key': 42})`.